diff --git a/src/jsdoccomment.js b/src/jsdoccomment.js
index 7a2d9a9cd79d8830ba30bcea5aa1e7716d3ea92f..d8b3da2b4f9ec957a2d27027292f7c63f5c4d0fa 100644
--- a/src/jsdoccomment.js
+++ b/src/jsdoccomment.js
@@ -398,7 +398,7 @@ const getJSDocComment = function (sourceCode, node, settings) {
         functionName === prevSibling.id.name) ||
       (prevSibling?.type === 'ExportNamedDeclaration' &&
         // @ts-expect-error Should be ok
-        prevSibling.declaration.type === 'TSDeclareFunction' &&
+        prevSibling.declaration?.type === 'TSDeclareFunction' &&
         // @ts-expect-error Should be ok
         prevSibling.declaration?.id?.name === functionName)
     ) {
